src/Interpreters/TensorFlowModel.cpp

```diff
@@ -0,0 +1,344 @@
+        if (model != nullptr)
+            api->TfLiteModelDelete(model);
+    }
+
+    ColumnPtr evaluate(const ColumnRawPtrs & columns) const override
+    {
+        if (columns.empty())
+            throw Exception("Got empty columns list for TensorFlow model", ErrorCodes::BAD_ARGUMENTS);
+        
+        if (columns.size() != static_cast<size_t>(tensors_count))
+        {
+            std::string msg;
+            {
+                WriteBufferFromString buffer(msg);
+                buffer << "Number of columns is different with number of tensors: ";
+                buffer << columns.size() << " vs " << tensors_count;
+            }
+            throw Exception(msg, ErrorCodes::BAD_ARGUMENTS);
+        }
+
+        switch(output_tensor_type)
+        {
+            case TfLiteType::kTfLiteFloat32: return evalImpl<ColumnFloat32, float>(columns);
+            case TfLiteType::kTfLiteFloat64: return evalImpl<ColumnFloat64, double>(columns);
+            case TfLiteType::kTfLiteInt8:    return evalImpl<ColumnInt8, int8_t>(columns);
+            case TfLiteType::kTfLiteInt32:   return evalImpl<ColumnInt32, int32_t>(columns);
+            case TfLiteType::kTfLiteInt64:   return evalImpl<ColumnInt64, int64_t>(columns);
+            case TfLiteType::kTfLiteUInt8:   return evalImpl<ColumnUInt8, uint8_t>(columns);
+            case TfLiteType::kTfLiteUInt32:  return evalImpl<ColumnUInt32, uint32_t>(columns);
+            case TfLiteType::kTfLiteUInt64:  return evalImpl<ColumnUInt64, uint64_t>(columns);
+            default:
+                throw Exception("Unsupported output tensor's data type", ErrorCodes::BAD_ARGUMENTS);
+        }
+    }
+
+    DataTypePtr getReturnType() const override
+    {
+        return returnType;
+    }
+
+private:
+    template<typename ColumnType, typename FieldType>
+    ColumnPtr evalImpl(const ColumnRawPtrs & columns) const
+    {
+        size_t rows_count = columns.front()->size();
+        auto res = ColumnType::create(rows_count);
+
+        for (size_t i = 0; i < rows_count; ++i)
+        {
+            for (int32_t j = 0; j < tensors_count; ++j)
+            {
+                auto tensor = api->TfLiteInterpreterGetInputTensor(interpreter, j);
+                if (tensor == nullptr)
+                    throw Exception("Tensor getting was failed", ErrorCodes::BAD_ARGUMENTS);
+
+                auto column = columns[j];
+                if (!column->isNumeric())
+                    throw Exception("Unsupported non-numeric data types", ErrorCodes::BAD_ARGUMENTS);
+
+                auto copyToTensor = [this, &tensor](const auto& value) {
+                    if (api->TfLiteTensorCopyFromBuffer(tensor, &value, sizeof(value)) != TfLiteStatus::kTfLiteOk) 
+                        throw Exception("Copy to tensor was failed", ErrorCodes::BAD_ARGUMENTS);
+                };
+
+                auto type = api->TfLiteTensorType(tensor);
+                switch(type)
```

Comment:
It seems that we copy into tensor by 1 element ? Better to get raw data from column and put it in one call.